device/flasher.go

```diff
@@ -101,30 +106,44 @@ func (d *flasher) PrepareForFlashing() error {
 
 	err = help.DeleteHost(filepath.Join((os.Getenv("HOME")), ".ssh", "known_hosts"), "localhost")
 	if err != nil {
-		logrus.Error(err)
+		log.Error(err)
 	}
 
-	fmt.Printf("[+] Uploading %s to virtual machine\n", zipName)
-	if err = d.conf.SSH.Scp(help.AddPathSuffix("unix", d.devRepo.Dir(), zipName), constants.TMP_DIR); err != nil {
-		return err
+	// check if zip is already inside VM
+	if _, eut, err := d.conf.SSH.Run("ls " + constants.TMP_DIR + zipName); err != nil || len(strings.TrimSpace(eut)) > 0 {
+		fmt.Printf("[+] Uploading %s to virtual machine\n", zipName)
+		if err = d.conf.SSH.Scp(help.AddPathSuffix(runtime.GOOS, d.devRepo.Dir(), zipName), constants.TMP_DIR); err != nil {
+			return err
+		}
+	} else {
+		log.Debug("Image zip exists inside VM")
 	}
-
-	fmt.Printf("[+] Extracting %s \n", zipName)
-	logrus.Debug("Extracting an image")
-	command := fmt.Sprintf(help.GetExtractCommand(zipName), help.AddPathSuffix("unix", constants.TMP_DIR, zipName), constants.TMP_DIR)
-	d.conf.SSH.SetTimer(help.SshExtendedCommandTimeout)
-	out, eut, err := d.conf.SSH.Run(command)
-	if err != nil || len(strings.TrimSpace(eut)) > 0 {
-		fmt.Println("[-] ", eut)
-		return err
+	if strings.HasSuffix(zipName, ".zip") {
+		if files, err := help.GetZipFiles(help.AddPathSuffix(runtime.GOOS, d.devRepo.Dir(), zipName)); err == nil && len(files) == 1 {
+			if _, eut, err := d.conf.SSH.Run("ls " + constants.TMP_DIR + files[0].Name); err == nil && len(strings.TrimSpace(eut)) == 0 {
+				log.Debug("Image file already extracted")
+				d.img = files[0].Name
+			}
+		}
 	}
 
-	logrus.Debug(out)
+	if d.img == "" {
+		fmt.Printf("[+] Extracting %s \n", zipName)
+		command := fmt.Sprintf(help.GetExtractCommand(zipName), help.AddPathSuffix("unix", constants.TMP_DIR, zipName), constants.TMP_DIR)
+		log.Debug("Extracting an image... ", command)
+		d.conf.SSH.SetTimer(help.SshExtendedCommandTimeout)
 
-	for _, raw := range strings.Split(out, " ") {
-		s := strings.TrimSpace(raw)
-		if s != "" && strings.HasSuffix(s, ".img") {
-			d.img = s
+		if out, eut, err := d.conf.SSH.Run(command); err != nil || len(strings.TrimSpace(eut)) > 0 {
+			fmt.Println("[-] ", eut)
+			return err
+		} else {
```

Comment:
`golint` suggests the established Go style. even if that isn't your preference... :-P

`golint` doesn't try to catch every possible situation. Your code isn't broken, it just doesn't follow the standard Go style. Which is something like:

```
v, ok := aMap[key]
if !ok {
  return
}
// use v
```